driver/generic/sendcommands.go

```diff
@@ -12,13 +14,21 @@ func (d *Driver) SendCommands(
 ) (*response.MultiResponse, error) {
 	d.Logger.Infof("SendCommands requested, sending '%s'", commands)
 
+	if len(commands) == 0 {
+		return nil, fmt.Errorf("%w: no commands provided", util.ErrNoOp)
+	}
+
 	op, err := NewOperation(opts...)
 	if err != nil {
 		return nil, err
 	}
 
 	m := response.NewMultiResponse(d.Transport.GetHost())
 
+	if len(commands) == 0 {
+		return m, err
+	}
+
```

Comment:
@carlmontanari 
unless I am missing something I don't see why this check is here, while you have it on Ln17 already?

Reply:
ahaha no, as per usual not you missing things -- just me :D ty ty, fixing and will merge!